File: windows/src/GoalMaker.Core/Planning/GoalDraft.cs
namespace GoalMaker.Core.Planning;

/// <summary>What a new or edited goal says (docs/goals.md); <see cref="PeriodStart"/> is moved to its period's first day.</summary>
public sealed record GoalDraft(
    string Title,
    GoalHorizon Horizon,
    DateOnly PeriodStart,
    string Mode = GoalRules.ModeDone,
    string? Emoji = null,
    string? ParentId = null,
    double? Target = null,
    string? Unit = null);

File: android/app/src/main/kotlin/com/goalmaker/app/application/planning/GoalDraft.kt
package com.goalmaker.app.application.planning

import java.time.LocalDate

/** What a new or edited goal says (docs/goals.md); [periodStart] is moved to its period's first day. */
data class GoalDraft(
    val title: String,
    val horizon: GoalHorizon,
    val periodStart: LocalDate,
    val mode: String = GoalRules.MODE_DONE,
    val emoji: String? = null,
    val parentId: String? = null,
    val target: Double? = null,
    val unit: String? = null,
)
